extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/graal/ByteBufferUnmapper.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.kafka.client.runtime.graal;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+
+import org.apache.kafka.common.utils.ByteBufferUnmapper;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import com.oracle.svm.core.jdk.JDK8OrEarlier;
+
+@TargetClass(value = ByteBufferUnmapper.class, onlyWith = JDK8OrEarlier.class)
+final class Target_org_apache_kafka_common_utils_ByteBufferUnmapper {
+
+    @Substitute
+    public static void unmap(String resourceDescription, ByteBuffer buffer) throws IOException {
+        if (!buffer.isDirect())
+            throw new IllegalArgumentException("Unmapping only works with direct buffers");
+        try {
+            Method cleanerMethod = buffer.getClass().getMethod("cleaner");
```

Comment:
Should we add ByteBuffer in the reflection list. The `cleaner` method _may_ be removed during the native compilation. The class returned by the "cleaner" method should also be registered.

Reply:
Done